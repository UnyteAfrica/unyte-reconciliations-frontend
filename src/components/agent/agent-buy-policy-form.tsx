import { useEffect, useRef, useState } from "react";
import { twMerge } from "tailwind-merge";
import { CustomInput } from "../shared/input";
import { Icon } from "../shared/icon";
import { MdOutlineFileUpload } from "react-icons/md";
import { formatToNaira } from "@/utils/utils";
import { SlidingButton } from "../shared/sliding-button";
import toast from "react-hot-toast";
import { useQuery } from "@tanstack/react-query";
import { AgentQueryKeys } from "@/utils/query-keys";
import { AgentApi } from "@/services/api/api-agent";
import { QuoteFormValue } from "@/types/types";
import { Loader } from "../loader";
import { produce } from "immer";

export const BuyPolicyForm: React.FC<{
  policyType: string;
  isOpen: boolean;
  policyCategory: string;
  onClose: () => void;
}> = ({ policyType, isOpen, onClose, policyCategory }) => {
  const containerRef = useRef<HTMLDivElement>(null);
  const [formPage, setFormPage] = useState(1);
  const [isSubmitted, setIsSubmitted] = useState(false);
  const [customerFormValues, setCustomerFormValues] = useState<
    QuoteFormValue[]
  >([]);
  const [insuranceFormValues, setInsuranceFormValues] = useState<
    QuoteFormValue[]
  >([]);

  const policyCategoryId = policyCategory.split(" ")[0].toLowerCase();

  const {
    isPending: isLoadingQuoteParams,
    data: quoteParamsData,
    // error: quoteParamsError,
  } = useQuery({
    queryKey: [AgentQueryKeys.quoteParams],
    queryFn: () => AgentApi.getQuoteParams(policyCategoryId),
  });

  useEffect(() => {
    setCustomerFormValues(quoteParamsData?.customer_metadata || []);
  }, [quoteParamsData?.customer_metadata]);

  useEffect(() => {
    setInsuranceFormValues(quoteParamsData?.insurance_details || []);
  }, [quoteParamsData?.insurance_details]);

  return (
    <div
      className={twMerge(
        "z-50 h-dvh w-screen fixed top-0 left-0 flex flex-col bg-white py-10 px-6 pb-6 transition translate-x-[100%] opacity-0 overflow-y-auto",
        isOpen && "translate-x-0 opacity-100"
      )}
      ref={containerRef}
    >
      <div className="flex justify-between items-center mb-4">
        <button
          onClick={(e) => {
            e.stopPropagation();
            if (isSubmitted) {
              setIsSubmitted(false);
              return;
            }
            if (formPage == 1) {
              onClose();
            } else {
              setFormPage(1);
            }
          }}
        >
          <Icon type="arrowBack" className="" />
        </button>
        {!isSubmitted && (
          <>
            <div
              className={twMerge(
                "bg-[#D9D9D9] rounded-full h-[6px] w-[40%] transition",
                formPage == 2 && "bg-[#25D366]"
              )}
            />
            <div className="bg-[#D9D9D9] rounded-full h-[6px] w-[40%]" />
          </>
        )}
      </div>
      {isSubmitted && <PolicyPayment />}
      {!isSubmitted && (
        <>
          <header className="mb-4">
            <h4 className="font-semibold mb-2">Sell {policyType}</h4>
            <em className="not-italic text-sm text-[#4F4F4F]">
              Enter customer's personal details
            </em>
          </header>
          <div>
            {isLoadingQuoteParams ? (
              <Loader className="mx-auto w-16 h-16 mt-20" />
            ) : formPage == 1 ? (
              <div className="space-y-6">
                {customerFormValues.map((formValue, idx) => (
                  <CustomInput
                    key={idx}
                    label={formValue.name}
                    labelClassName="text-sm"
                    className="h-14 border-[#e0e0e0] text-sm"
                    value={formValue.value}
                    onChange={(e) =>
                      setCustomerFormValues(
                        produce((draft) => {
                          draft[idx].value = e.target.value;
                        })
                      )
                    }
                  />
                ))}
              </div>
            ) : (
              formPage == 2 && (
                <div className="space-y-6">
                  {insuranceFormValues.map((formValue, idx) => (
                    <CustomInput
                      key={idx}
                      label={formValue.name}
                      labelClassName="text-sm"
                      className="h-14 border-[#e0e0e0] text-sm"
                    />
                  ))}

                  <button className="w-full rounded-lg bg-[#F2F2F2] text-[#333] flex items-center justify-center py-4">
                    <MdOutlineFileUpload className="text-4xl inline-block mr-2" />
                    <span className="font-medium text-sm">
                      Upload Passport Data Page
                    </span>
                  </button>
                  <div className="flex justify-center items-center mt-4">
                    <Icon type="verified" className="inline-block mr-1" />
                    <em className="not-italic text-[#333] text-sm">
                      Your data is safe and secure
                    </em>
                  </div>
                </div>
              )
            )}
          </div>
          <div className="grow min-h-10" />
          <button
            onClick={(e) => {
              e.preventDefault();
              if (formPage == 1) {
                setFormPage(2);
                containerRef.current?.scrollTo({
                  top: 0,
                  left: 0,
                  behavior: "smooth",
                });
              } else {
                //Submit Form
                setIsSubmitted(true);
              }
            }}
            className="w-full bg-[#25D366] py-4 text-lg font-medium rounded-lg text-white"
          >
            {formPage == 1 ? "Continue" : "Get Quote"}
          </button>
          <em className="block text-center not-italic text-xs mt-10">
            Powered by Unyte Africa
          </em>
        </>
      )}
    </div>
  );
};

const PolicyPayment = () => {
  return (
    <div className="py-6 bg-white">
      <div className="w-full bg-[#C8FFC6] flex justify-center items-center mb-3 py-6 rounded-lg">
        <Icon type="travel" className="w-[100px] h-[100px]" />
      </div>
      <div className="py-[10px] mx-auto w-[165px] bg-[#101323] flex justify-center items-center rounded-full mb-3">
        <Icon type="whiteReceipt" className="mr-1 inline-block" />
        <span className="text-xs font-medium text-white">
          View coverage details
        </span>
      </div>
      <div className="bg-[#F9FAFB] rounded-lg p-4 space-y-4 mb-3">
        <div className="flex justify-between items-center">
          <em className="not-italic text-sm text-[#667085]">Policy number</em>
          <em className="not-italic text-sm font-medium text-[#101323]">
            **** **** 5678
          </em>
        </div>
        <div className="flex justify-between items-center">
          <em className="not-italic text-sm text-[#667085]">Effective date</em>
          <em className="not-italic text-sm font-medium text-[#101323]">
            19/09/2024
          </em>
        </div>
        <div className="flex justify-between items-center">
          <em className="not-italic text-sm text-[#667085]">Expiration Date</em>
          <em className="not-italic text-sm font-medium text-[#101323]">
            11/12/2024
          </em>
        </div>
        <div className="flex justify-between items-center">
          <em className="not-italic text-sm text-[#667085]">Cost</em>
          <em className="not-italic text-sm font-medium text-[#101323]">
            {formatToNaira(7500)}
          </em>
        </div>
      </div>
      <div className="space-y-[10px] bg-[#F9FAFB] rounded-lg p-4 mb-4">
        <em className="not-italic font-medium block">
          AXA Mansard Insurance LTD
        </em>
        <em className="not-italic font-medium block">3112456789</em>
        <em className="not-italic font-medium block">GTBank</em>
      </div>
      <em className="block mb-4 uppercase text-center not-italic">or</em>
      <Icon type="qr" className="mb-8 mx-auto" />
      <SlidingButton onSlide={() => toast.success("Policy Sold")} />
    </div>
  );
};
