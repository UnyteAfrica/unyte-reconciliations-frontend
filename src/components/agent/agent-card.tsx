import { Agent, AgentStatus } from "@/types/types";
import { Icon } from "../shared/icon";
import { formatToNaira, getCompanyInitials } from "@/utils/utils";
import { twMerge } from "tailwind-merge";

export const AgentCard: React.FC<{ agent: Agent; onClose?: () => void }> = ({
  onClose,
  agent,
}) => {
  return (
    <div className="bg-white p-10 h-screen w-[720px] overflow-y-auto">
      <div className="flex justify-between items-center mb-6">
        <h1 className="text-[32px] font-semibold">Agent Information</h1>
        <button onClick={onClose}>
          <Icon type="cancel" className="w-3 h-3" />
        </button>
      </div>
      <div className="bg-[#F2F2F2] py-4 px-8 rounded-2xl flex items-center mb-8">
        <div
          style={{
            background:
              "linear-gradient(163.115deg, #C8FFC6 0%, rgba(252, 230, 26, 10%) 62%), linear-gradient(#5CC758, #5CC758)",
          }}
          className="w-[102px] h-[102px] rounded-full flex justify-center items-center text-white mr-4 text-[32px] font-semibold"
        >
          {getCompanyInitials(agent?.name)}
        </div>
        <div>
          <div className="flex items-center">
            <h3 className="text-2xl font-medium mr-4 text-[#333]">
              {agent.name}
            </h3>
            <div
              className={twMerge(
                "w-[84px] h-[24px] text-white flex justify-center items-center rounded font-medium",
                agent.agentStatus == AgentStatus.Active && "bg-[#25d366]",
                agent.agentStatus == AgentStatus.Inactive && "bg-[#D32525]"
              )}
            >
              {agent.agentStatus}
            </div>
          </div>
          <a
            href={`mailto:${agent.email}`}
            className="text-xl font-medium text-[#25d366] underline"
          >
            {agent.email}
          </a>
        </div>
      </div>
      <div className="flex justify-between items-stretch mb-8">
        <div className="w-[201px]  bg-[#f2f2f2] rounded-2xl px-8 py-9 flex flex-col">
          <em className="not-italic font-medium text-[#4f4f4f] mb-4">
            Policies Sold
          </em>
          <em className="not-italic font-medium text-xl text-[#333]">
            {agent.policiesSold} policies
          </em>
        </div>
        <div className="w-[201px]  bg-[#f2f2f2] rounded-2xl px-8 py-9 flex flex-col">
          <em className="not-italic font-medium text-[#4f4f4f] mb-4">
            Commission Total
          </em>
          <em className="not-italic font-medium text-xl text-[#333]">
            {formatToNaira(agent.commissions)}
          </em>
        </div>
        <div className="w-[201px]  bg-[#f2f2f2] rounded-2xl pl-8 py-9 flex flex-col">
          <em className="not-italic font-medium text-[#4f4f4f] mb-4">
            Date Added
          </em>
          <em className="not-italic font-medium text-xl text-[#333]">
            {agent.dateAdded}
          </em>
        </div>
      </div>
      <div className="bg-[#f2f2f2] rounded-2xl flex flex-col p-8 space-y-2 mb-8">
        <em className="not-italic text-2xl text-[#4f4f4f]">{agent.name}</em>
        <em className="not-italic text-2xl text-[#333]">3324156708</em>
        <em className="not-italic text-2xl text-[#333]">
          Moniepoint Microfinance Bank
        </em>
      </div>
      <div className="bg-[#f2f2f2] rounded-2xl flex flex-col p-8 space-y-4">
        <h2 className="font-medium text-2xl">Commission Activity</h2>
        <div className="flex justify-between items-center">
          <em className="not-italic text-2xl text-[#4f4f4f]">May 2023</em>
          <em className="not-italic font-medium text-2xl text-[#333]">
            {formatToNaira(5000)}
          </em>
        </div>
        <div className="flex justify-between items-center">
          <em className="not-italic text-2xl text-[#4f4f4f]">June 2023</em>
          <em className="not-italic font-medium text-2xl text-[#333]">
            {formatToNaira(10000)}
          </em>
        </div>
        <div className="flex justify-between items-center">
          <em className="not-italic text-2xl text-[#4f4f4f]">July 2023</em>
          <em className="not-italic font-medium text-2xl text-[#333]">
            {formatToNaira(6000)}
          </em>
        </div>
      </div>
    </div>
  );
};
