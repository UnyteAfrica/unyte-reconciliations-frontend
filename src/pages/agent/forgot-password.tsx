import { CustomInput } from "@/components/shared/input";
import { z } from "zod";
import { useForm } from "react-hook-form";
import { Link } from "react-router-dom";
import { BrowserComboRoutes } from "@/utils/routes";
import { zodResolver } from "@hookform/resolvers/zod";
import { useMutation } from "@tanstack/react-query";
import { MutationKeys } from "@/utils/mutation-keys";
import { agentForgotPassword } from "@/api/api-agent";
import { Loader } from "@/components/loader";
import toast from "react-hot-toast";

const formSchema = z.object({
  email: z.string().email("Invalid email entered"),
});

export const AgentForgotPasswordPage = () => {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<z.infer<typeof formSchema>>({
    resolver: zodResolver(formSchema),
    defaultValues: {
      email: "",
    },
  });

  const { mutate: mForgotPassword, isPending: isLoadingForgotPassword } =
    useMutation({
      mutationKey: [MutationKeys.agentForgotPassword],
      mutationFn: (email: string) => agentForgotPassword(email),
      onSuccess: (data) => {
        console.log(data);
        toast.success(data.data.message);
      },
    });

  const onSubmit = (data: z.infer<typeof formSchema>) => {
    console.log(data);
    mForgotPassword(data.email);
  };

  return (
    <div className="flex justify-center items-center  bg-[#f5f5f5] min-h-screen">
      <div className="p-10 bg-white flex justify-center items-center w-[700px] h-[500px] rounded-2xl">
        <form
          onSubmit={handleSubmit(onSubmit)}
          className="text-[#333] w-[560px]"
        >
          <header>
            <p className="text-center mb-11 font-medium text-[28px]">
              Forgot Password
            </p>
          </header>
          <div className="space-y-6">
            <CustomInput
              label="Email"
              placeholder="johndoe@gmail.com"
              error={errors.email?.message?.toString()}
              {...register("email")}
            />
            <div>
              <p className="mb-2">
                Already have an account?{" "}
                <Link
                  to={BrowserComboRoutes.agentLogin}
                  className="text-primary"
                >
                  Log In
                </Link>
              </p>
              <button
                className="w-full font-medium text-xl leading-[24px] bg-primary h-[72px] text-white rounded-2xl"
                disabled={isLoadingForgotPassword}
              >
                {isLoadingForgotPassword ? (
                  <Loader className="mx-auto" />
                ) : (
                  "Reset Password"
                )}
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  );
};
