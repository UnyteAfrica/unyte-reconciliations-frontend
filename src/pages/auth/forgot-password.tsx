import { CustomInput } from "@/components/shared/input";
import { z } from "zod";
import { useForm } from "react-hook-form";
import { Link } from "react-router-dom";
import { BrowserRoutes } from "@/utils/routes";
import { zodResolver } from "@hookform/resolvers/zod";
import { useMutation } from "@tanstack/react-query";
import { MutationKeys } from "@/utils/mutation-keys";
import { Loader } from "@/components/loader";
import toast from "react-hot-toast";
import { useMediaQuery } from "@/utils/hooks";
import { Icon } from "@/components/shared/icon";
import { logger } from "@/utils/logger";
import { forgotPassword } from "@/services/api/api-base";

const formSchema = z.object({
  email: z.string().email("Invalid email entered"),
});

export const ForgotPasswordPage = () => {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<z.infer<typeof formSchema>>({
    resolver: zodResolver(formSchema),
    defaultValues: {
      email: "",
    },
  });

  const { mutate: mForgotPassword, isPending: isLoadingForgotPassword } =
    useMutation({
      mutationKey: [MutationKeys.agentForgotPassword],
      mutationFn: (email: string) => forgotPassword(email),
      onSuccess: (data) => {
        logger.log(data);
        toast.success(data.data.message);
      },
    });

  const onSubmit = (data: z.infer<typeof formSchema>) => {
    logger.log(data);
    mForgotPassword(data.email);
  };

  const { isMediaQueryMatched } = useMediaQuery(1024);

  if (!isMediaQueryMatched)
    return (
      <div className="px-5 py-10 max-w-[600px] min-h-svh mx-auto flex flex-col ">
        <Icon type="logo" className="mb-6 block w-28" />
        <div className="grow" />
        <form onSubmit={handleSubmit(onSubmit)}>
          <header>
            <p className="mb-2 font-semibold text-[20px] text-center">
              Forgot Password
            </p>
          </header>
          <div className="space-y-6">
            <CustomInput
              label="Email"
              placeholder="johndoe@gmail.com"
              labelClassName="text-sm text-[#333"
              className="p-2 h-[58px] border-[#E0E0E0]"
              error={errors.email?.message?.toString()}
              {...register("email")}
            />
            <div>
              <p className="mb-2 text-sm">
                Already have an account?{" "}
                <Link to={BrowserRoutes.login} className="text-mPrimary">
                  Log In
                </Link>
              </p>
              <button
                className="w-full font-medium text-xl leading-[24px] bg-mPrimary h-[58px] text-white rounded-2xl"
                disabled={isLoadingForgotPassword}
              >
                {isLoadingForgotPassword ? (
                  <Loader className="mx-auto" />
                ) : (
                  "Reset Password"
                )}
              </button>
            </div>
          </div>
        </form>
        <div className="grow" />
      </div>
    );

  return (
    <div className="flex justify-center items-center  bg-[#f5f5f5] min-h-screen">
      <div className="p-10 bg-white flex justify-center items-center w-[700px] h-[500px] rounded-2xl">
        <form
          onSubmit={handleSubmit(onSubmit)}
          className="text-[#333] w-[560px]"
        >
          <header>
            <p className="text-center mb-11 font-medium text-[28px]">
              Forgot Password
            </p>
          </header>
          <div className="space-y-6">
            <CustomInput
              label="Email"
              placeholder="johndoe@gmail.com"
              error={errors.email?.message?.toString()}
              {...register("email")}
            />
            <div>
              <p className="mb-2">
                Already have an account?{" "}
                <Link to={BrowserRoutes.login} className="text-mPrimary">
                  Log In
                </Link>
              </p>
              <button
                className="w-full font-medium text-xl leading-[24px] bg-mPrimary h-[72px] text-white rounded-2xl"
                disabled={isLoadingForgotPassword}
              >
                {isLoadingForgotPassword ? (
                  <Loader className="mx-auto" />
                ) : (
                  "Reset Password"
                )}
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  );
};
