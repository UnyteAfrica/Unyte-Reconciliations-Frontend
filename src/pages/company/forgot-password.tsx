import { CustomInput } from "@/components/shared/input";
import { z } from "zod";
import { useForm } from "react-hook-form";
import { Link } from "react-router-dom";
import { BrowserComboRoutes } from "@/utils/routes";
import { zodResolver } from "@hookform/resolvers/zod";

const formSchema = z.object({
  email: z.string().email("Invalid email entered"),
});

export const CompanyForgotPasswordPage = () => {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<z.infer<typeof formSchema>>({
    resolver: zodResolver(formSchema),
    defaultValues: {
      email: "",
    },
  });

  const onSubmit = (data: z.infer<typeof formSchema>) => {
    console.log(data);
  };

  return (
    <div className="flex justify-center items-center  bg-[#f5f5f5] min-h-screen">
      <div className="p-10 bg-white flex justify-center items-center w-[700px] h-[500px] rounded-2xl">
        <form
          onSubmit={handleSubmit(onSubmit)}
          className="text-[#333] w-[560px]"
        >
          <header>
            <p className="text-center mb-11 font-medium text-[28px]">
              Company Forget Password
            </p>
          </header>
          <div className="space-y-6">
            <CustomInput
              label="Email"
              placeholder="johndoe@gmail.com"
              error={errors.email?.message?.toString()}
              {...register("email")}
            />
            <div>
              <p className="mb-2">
                Already have an account?{" "}
                <Link
                  to={BrowserComboRoutes.companyLogin}
                  className="text-primary"
                >
                  Log In
                </Link>
              </p>
              <button className="w-full font-medium text-xl leading-[24px] bg-primary h-[72px] text-white rounded-2xl">
                Reset Password
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  );
};
