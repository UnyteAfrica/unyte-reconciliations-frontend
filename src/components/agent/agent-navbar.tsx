import { useContext, useState } from "react";
import { NavLink, useLocation } from "react-router-dom";
import { BiChevronDown } from "react-icons/bi";

import { Icon } from "../shared/icon";
import { BrowserComboRoutes, BrowserRoutes } from "@/utils/routes";
import { cx } from "class-variance-authority";
import { twMerge } from "tailwind-merge";
import { clearCredentials } from "@/utils/utils";
import { UserType } from "@/types/types";
import { AgentContext } from "@/context/agent.context";
import { getAgentDetails } from "@/services/api/api-agent";
import { AgentQueryKeys } from "@/utils/query-keys";
import { useQuery } from "@tanstack/react-query";
import { Loader } from "../loader";

type UrlLink = {
  text: string;
  url: string;
};

const navLinks: UrlLink[] = [
  {
    text: "Overview",
    url: BrowserComboRoutes.agentDashboard + BrowserRoutes.overview,
  },
  {
    text: "Policies",
    url: BrowserComboRoutes.agentDashboard + BrowserRoutes.policies,
  },
  {
    text: "Commissions",
    url: BrowserComboRoutes.agentDashboard + BrowserRoutes.commissions,
  },
  {
    text: "Devices",
    url: BrowserComboRoutes.agentDashboard + BrowserRoutes.devices,
  },
];

export const AgentNavbar = () => {
  const [isOpen, setIsOpen] = useState(false);
  const location = useLocation();
  const { setIsLoggedIn } = useContext(AgentContext);

  const { data: agentDetailsData, isPending: isAgentDetailsLoading } = useQuery(
    {
      queryKey: [AgentQueryKeys.details],
      queryFn: () => getAgentDetails(),
    }
  );

  const toggleDropdown = () => {
    setIsOpen(!isOpen);
  };

  const getInitials = (firstName: string, lastName: string) =>
    firstName.substring(0, 1).toUpperCase() +
    lastName.substring(0, 1).toUpperCase();

  const agentDetails = agentDetailsData?.data;

  return (
    <>
      <div
        id="navbar-container"
        className="flex justify-between items-center p-4 mx-12"
      >
        <div>
          <Icon className="shrink-0" type="logo" alt="unyte logo" />
        </div>
        <div className="space-x-24">
          {navLinks.map((navLink, idx) => (
            <NavLink
              key={idx}
              to={navLink.url}
              className={({ isActive }) =>
                cx(
                  "font-semibold text-lg duration-300",
                  (isActive ||
                    location.pathname.includes(navLink.text.toLowerCase())) &&
                    "text-[#25D366] underline underline-offset-[30px] decoration-4"
                )
              }
            >
              {navLink.text}
            </NavLink>
          ))}
        </div>

        {isAgentDetailsLoading ? (
          <Loader />
        ) : (
          !!agentDetails && (
            <div id="profile">
              <div className="relative">
                <button
                  className="px-4 py-2 text-[#333333] rounded-md"
                  onClick={toggleDropdown}
                >
                  <div className="space-x-2 flex flex-row items-center">
                    <div className="rounded-full h-10 w-10 p-2 bg-gray-200 text-base flex items-center justify-center">
                      {getInitials(
                        agentDetails.first_name,
                        agentDetails.last_name
                      )}
                    </div>
                    <span className="text-lg font-semibold max-xl:hidden">
                      {agentDetails.first_name + " " + agentDetails.last_name}
                    </span>{" "}
                    <BiChevronDown
                      className={cx(
                        "transition-all duration-300",
                        isOpen && "rotate-180"
                      )}
                    />
                  </div>
                </button>
                <div
                  className={twMerge(
                    "absolute w-full top-[65px] right-0 mt-2 py-2 z-10 border border-[#e1e1e1] bg-white shadow-lg rounded-lg max-h-0 transition-all duration-300 overflow-hidden opacity-0",
                    isOpen && "max-h-[240px] opacity-100"
                  )}
                >
                  <div className="px-6 py-4 flex items-center p-2">
                    <div className="rounded-full h-10 w-10 p-2 bg-gray-200 text-base flex items-center justify-center mr-2">
                      {getInitials(
                        agentDetails.first_name,
                        agentDetails.last_name
                      )}
                    </div>
                    <span className="text-base text-[#333] font-medium">
                      {agentDetails.first_name + " " + agentDetails.last_name}
                    </span>
                  </div>
                  <hr />
                  <NavLink
                    to={
                      BrowserComboRoutes.agentDashboard + BrowserRoutes.profile
                    }
                    className="block px-6 py-4 text-[#333] font-medium"
                  >
                    Profile
                  </NavLink>
                  <hr />
                  <button
                    className="block px-6 py-4 text-[#EB5757] font-medium"
                    onClick={() => {
                      clearCredentials(UserType.agent);
                      setIsLoggedIn(false);
                    }}
                  >
                    Sign out
                  </button>
                </div>
              </div>
            </div>
          )
        )}
      </div>
      <hr />
    </>
  );
};
