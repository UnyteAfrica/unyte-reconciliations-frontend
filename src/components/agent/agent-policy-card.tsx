import { BuyPolicyForm } from "@/components/agent/agent-buy-policy-form";
import { useEffect, useState } from "react";
import { createPortal } from "react-dom";

export const PolicyCard: React.FC<{
  product: string;
  policyCategory: string;
}> = ({ product, policyCategory }) => {
  const [isOpen, setIsOpen] = useState(false);

  useEffect(() => {
    if (isOpen) {
      document.body.style.overflow = "hidden";
    } else {
      document.body.style.overflow = "unset";
    }
  }, [isOpen]);

  return (
    <div
      onClick={() => setIsOpen(true)}
      className="bg-white rounded-2xl p-4 cursor-pointer"
    >
      {createPortal(
        <BuyPolicyForm
          isOpen={isOpen}
          policyType={product}
          policyCategory={policyCategory}
          onClose={() => setIsOpen(false)}
        />,
        document.body
      )}

      <div className="flex">
        <div>
          <h3 className="not-italic text-sm font-medium inline-block mb-1 text-[#101323]">
            {product}
          </h3>
        </div>
      </div>
    </div>
  );
};
