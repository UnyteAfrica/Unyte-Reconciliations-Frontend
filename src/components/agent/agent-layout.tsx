import { Outlet } from "react-router-dom";
import { AgentNavbar } from "./agent-navbar";

export const AgentLayout = () => {
  return (
    <div>
      <AgentNavbar />
      <Outlet />
    </div>
  );
};
