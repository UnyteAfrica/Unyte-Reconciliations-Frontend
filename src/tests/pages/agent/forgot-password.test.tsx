import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import React from "react";
import { AgentForgotPasswordPage } from "@/pages/agent/forgot-password";

function setup(reactNode: React.ReactNode) {
  return {
    user: userEvent.setup(),
    ...render(reactNode),
  };
}

describe("Agent Forgot Password Page", () => {
  it("renders correctly", () => {
    const tree = render(
      <BrowserRouter>
        <AgentForgotPasswordPage />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it("should validate email", async () => {
    const { user } = setup(
      <BrowserRouter>
        <AgentForgotPasswordPage />
      </BrowserRouter>
    );

    const emailInput = screen.getByLabelText("Email");
    await user.click(emailInput);
    await user.keyboard("johndoe");
    const submitButton = screen.getByText("Reset Password");
    await user.click(submitButton);
    const errorMessage = screen.getByText("Invalid email entered");
    expect(errorMessage).toBeVisible();
  });
});
