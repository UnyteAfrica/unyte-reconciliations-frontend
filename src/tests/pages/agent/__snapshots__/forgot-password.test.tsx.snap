// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Agent Forgot Password Page > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="px-5 py-10 max-w-[600px] min-h-screen mx-auto flex flex-col "
      >
        <img
          alt=""
          class="mb-6 block w-28"
          src="/src/assets/Icons/UnyteLogo.svg"
        />
        <div
          class="grow"
        />
        <form>
          <header>
            <p
              class="mb-2 font-semibold text-[20px]"
            >
              Forgot Password
            </p>
          </header>
          <div
            class="space-y-6"
          >
            <div
              class=""
            >
              <label
                class="mb-2 block text-sm text-[#333"
                for="email"
              >
                Email
                 
              </label>
              <input
                class="block text-lg font-sora w-full border rounded-lg outline-none focus:border-black p-2 h-[58px] border-[#E0E0E0]"
                id="email"
                name="email"
                placeholder="johndoe@gmail.com"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div>
              <p
                class="mb-2 text-sm"
              >
                Already have an account?
                 
                <a
                  class="text-primary"
                  href="/agent/login"
                >
                  Log In
                </a>
              </p>
              <button
                class="w-full font-medium text-xl leading-[24px] bg-primary h-[58px] text-white rounded-2xl"
              >
                Reset Password
              </button>
            </div>
          </div>
        </form>
        <div
          class="grow"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="px-5 py-10 max-w-[600px] min-h-screen mx-auto flex flex-col "
    >
      <img
        alt=""
        class="mb-6 block w-28"
        src="/src/assets/Icons/UnyteLogo.svg"
      />
      <div
        class="grow"
      />
      <form>
        <header>
          <p
            class="mb-2 font-semibold text-[20px]"
          >
            Forgot Password
          </p>
        </header>
        <div
          class="space-y-6"
        >
          <div
            class=""
          >
            <label
              class="mb-2 block text-sm text-[#333"
              for="email"
            >
              Email
               
            </label>
            <input
              class="block text-lg font-sora w-full border rounded-lg outline-none focus:border-black p-2 h-[58px] border-[#E0E0E0]"
              id="email"
              name="email"
              placeholder="johndoe@gmail.com"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div>
            <p
              class="mb-2 text-sm"
            >
              Already have an account?
               
              <a
                class="text-primary"
                href="/agent/login"
              >
                Log In
              </a>
            </p>
            <button
              class="w-full font-medium text-xl leading-[24px] bg-primary h-[58px] text-white rounded-2xl"
            >
              Reset Password
            </button>
          </div>
        </div>
      </form>
      <div
        class="grow"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
  "user": {
    "clear": [Function],
    "click": [Function],
    "copy": [Function],
    "cut": [Function],
    "dblClick": [Function],
    "deselectOptions": [Function],
    "hover": [Function],
    "keyboard": [Function],
    "paste": [Function],
    "pointer": [Function],
    "selectOptions": [Function],
    "setup": [Function],
    "tab": [Function],
    "tripleClick": [Function],
    "type": [Function],
    "unhover": [Function],
    "upload": [Function],
  },
}
`;
